packages/xo-server/src/api/vm.mjs

```diff
@@ -173,7 +197,41 @@ export const create = defer(async function ($defer, params) {
   }
 
   if (params.bootAfterCreate) {
-    ignoreErrors.call(xapi.startVm(vm._xapiId))
+    try {
+      const start = new Date()
+      await xapi.startVm(vm._xapiId)
+
+      if (cloudConfigVdiUuid !== undefined && params.destroyCloudConfigVdiAfterBoot) {
+        const started = new Date()
+        const timeout = 10 * 60 * 1000
+        const startDuration = started - start
+
+        let remainingTimeout = timeout - startDuration
+
+        // wait for the 'Running' event to be really stored in local xapi object cache
+        await xapi.waitObjectState(vm.$ref, vm => vm.power_state === 'Running', {
+          timeout: remainingTimeout,
+        })
+
+        const running = new Date()
+        remainingTimeout -= running - started
+
+        if (remainingTimeout < 0) {
+          log.warn(`local xapi did not get runnig state for VM ${vm.id} after ${timeout / 1000} seconds`)
+          return vm.id
+        }
+
+        // wait for the guest tool version to be defined
+        await xapi.waitObjectState(vm.guest_metrics, gm => gm?.PV_drivers_version?.major !== undefined, {
+          timeout: remainingTimeout,
+        })
```

Comment:
I think this part may be moved to the `startVm` method in `xo-server/srx/xapi` , with two options : `waitForRunning` and `waitForXenTool`. If we're not sure if the VM has the xentools installed, we should only use `waitForRunning` + a security delay. The delay should be configurable or at least shown in the label 

I don't think we should await it since it stalls the create form as long as the start is not done